backend/synthesis/synthesis.py

```diff
@@ -35,20 +35,40 @@ def syntheize_modules(data: dict, zipfile: InMemoryZip) -> Tuple[InMemoryZip, Di
     synhronize_frequency = {}
     optional_files = {}
 
+    dep_no = {} # Dictionary to store the number of blocks of each type
+
     for key, dependency in data['dependencies'].items():
+
         components = dependency['design']['graph']['blocks']
+
         for block in components:
             if block['type'] == 'basic.code':
                 # If code, generate python file.
                 script = block['data']['code']
-                script_name = dependency['package']['name']
+                # The file name is stored in script_name here
+                script_name = dependency['package']['name'] 
                 synhronize_frequency[key] = get_number_or_default(block['data'].get('frequency', 30), 30)
                 # Mark the optional files required by the current block, if needed.
                 if script_name in OPTIONAL_FILES:
                     optional_files[script_name] = True
-                script_name += dependency['package']['version'].replace('.', '')
+
+                # The version name is simply added to the name of the file alongwith an underscore
+                script_name += dependency['package']['version'].replace('.', '') + "_"
+
+                # If the block already exists, then add 1 to the number to be appended to it 
+                if script_name in dep_no:
+                    dep_no[script_name] += 1
+                else:
+                    dep_no[script_name] = 1
+
+                # Replace the underscore with the appropraite block number
+                script_name = script_name.replace("_", str(dep_no[script_name]))
```

Comment:
Any particular reason to use this logic of replacing underscores with the count? Instead of just appending to the end?
If the script name itself has an underscore, that would be replaced as well. Though it might not cause any runtime issues, we will see weird file names in that case.

Reply:
I'd actually tried a couple of different approaches before and one of them required this kind of replacement, however, the current one will work best without the replacement. I'll change it now